### PR TITLE
chore: biome linter migration

### PR DESCRIPTION
**Pull request by DanielSinclair**
## Changes
- Deprecated `babel`, `estlint`, `prettier` and their accompanying patches/plugins and configs
- Adopted Biome with `biome.json` configuration borrowed from Wagmi project
- Altered `lint` `lint:fix` `format:check` `format:fix` commands for Biome
- Removed explicit `allowedDuplicatePackages` pnpm configuration for eslint mismatches

## Style changes
- Trailing commas are required
- Ignoring formatter complaints for design system tokens with keys declared with quotes
- Forcing `const` in more places
- Forcing `'` quotes in more places over templated string literals
- Forcing `(value) => {}` arrow function syntax

### CODE REVIEW

File: biome.json [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        "useGroupedTypeImport": "off"
+      },
+      "performance": {
+        "noDelete": "off"
+      },
+      "style": {
+        "noNonNullAssertion": "off",
+        "useShorthandArrayType": "error",
+        "useSingleVarDeclarator": "off"
+      },
```
Comment (KosmosKey) [inline biome.json:39]: Looks really good 🙏. I don't have any feedbacks on the `biome.json` file apart form maybe having : `"noDuplicateParameters": "off"` ? [Link](https://biomejs.dev/linter/rules/no-duplicate-parameters)

File: package.json [hunk 2]
```diff
@@ -13,10 +13,10 @@
...
     "build": "pnpm --recursive --parallel --filter=!rainbowkit-next-app --filter=!generated-test-app build",
     "start": "pnpm dev",
-    "lint": "eslint . --ext=js,jsx,mjs,ts,cjs,tsx && pnpm format:check && pnpm --recursive typecheck",
-    "lint:fix": "eslint . --ext=js,jsx,mjs,ts,cjs,tsx --fix && pnpm format:fix",
-    "format:check": "prettier --check \"./**/*.{ts,js,jsx,mjs,cjs,md,mdx,tsx}\"",
-    "format:fix": "prettier --write \"./**/*.{ts,js,jsx,mjs,cjs,md,mdx,tsx}\"",
+    "lint": "biome lint . && pnpm format:check && pnpm --recursive typecheck",
+    "lint:fix": "biome lint . --apply && pnpm format:fix",
+    "format:check": "biome check .",
+    "format:fix": "biome format . --write",
```
Comment (KosmosKey) [inline package.json:19]: Great scripts 🙏

File: packages/rainbowkit/src/components/Dialog/FocusTrap.tsx [hunk 3]
```diff
@@ -23,6 +23,7 @@ export function FocusTrap(props: JSX.IntrinsicElements['div']) {
     };
   }, []);
 
+  // biome-ignore lint/nursery/useExhaustiveDependencies: TODO
```
Comment (KosmosKey) [inline packages/rainbowkit/src/components/Dialog/FocusTrap.tsx:26]: This can stay as it is for now 🙏

**Review by 0xmkh — APPROVED** [review]
This is a great PR. Choosing biomejs rather than eslint was a great decision. Biomejs is actually 5x faster than eslint. This one will be exciting 🙏

image

File: site/lib/prism-diff-highlight.js [hunk 4]
```diff
@@ -47,32 +46,33 @@
     // one of the diff tokens without any nested tokens
     if (env.type in PREFIXES) {
       /** @type {string} */
-      var content = env.content.replace(HTML_TAG, ''); // remove all HTML tags
+      const content = env.content.replace(HTML_TAG, ''); // remove all HTML tags
```
[the remark this answers is not in the document]
Comment (KosmosKey) [reply site/lib/prism-diff-highlight.js:49]: This seems like a false positive check. It's a REGEX so i can't see any issues with it.

image

**Pull request merged**